### DISCUSSION TITLE
Normalization range

### DISCUSSION THREAD

**Discussion by talhaanwarch** [albumentations-team/albumentations#953] [Q&A]
I am confused whether albumentation normalize between 0 and 1 or between -1 and 1. I tested it is between -1 and 1, but I assume it to be between 0 and 1. 
Here is code
```
aug= A.Compose([
            A.Resize(224, 224),
            #CenterCrop(224,224),
            A.HorizontalFlip(p=0.5),
            A.VerticalFlip(p=0.5),
            A.ShiftScaleRotate(0.05,0.05,5), 
            A.Cutout(p=0.5),
            A.Normalize(mean=(0.5), std=(0.5)),
            ToTensorV2(p=1.0),
        ], p=1.0)
```

**Comment by Dipet** [comment]
Try to use this: `A.Compose([A.Normalize(mean=0.5, std=1.0)])` to normalize image to range `[0, 1]`

**Comment by FreeFly19** [comment]
The most common values:
`A.Normalize(mean=0.0, std=1.0)` -> `[0, 1]`
`A.Normalize(mean=0.5, std=0.5)` -> `[-1, 1]`

Just for understanding:
`A.Normalize(mean=0.5, std=1.0)` -> `[-0.5, 0.5]`
`A.Normalize(mean=1.0, std=1.0)` -> `[-1, 0]`

Here is the formula from the source code:
`img = (img - mean * max_pixel_value) / (std * max_pixel_value)`

**Reply by Dipet** [reply]
Yes, you're right, my fault.

**Reply by STHxiao** [reply]
Hi, why need to multiply by `max_pixel_value`, normally standardized formulas don't have this item, looking forward to your reply!

**Reply by Foundsheep** [reply]
It seems like, it's because 
1. the range of pixels in the image could vary between [0, 1] and [0, 255] etc.
2. the mean and std value such as 0.5, 1.0 are too small if used as a absolute value with range [0, 255]
3. therefore, to use the scale of the range, we need to multiply by `max_pixel_value`

These are my thoughts derived from a little bit of calculations

**Reply by STHxiao** [reply]
I guess it's not like that, it seems like it's because the mean on the default ImageNet dataset is calculated by dividing by 255, so it needs to be multiplied back when normalizing. You can refer to the default parameters of this API.
